beacon/src/avahi/mod.rs

```diff
@@ -34,13 +36,10 @@ impl<'a> Avahi<'a> {
         Ok(browsing::Browsing(proxy.receive_item_new().await?))
     }
 
-    pub async fn resolve(
-        &self,
-        service: browsing::Service,
-    ) -> Result<resolving::Resolving, zbus::Error> {
-        let path = self
+    pub async fn resolve(&self, service: browsing::Service) -> Result<Resolved, zbus::Error> {
+        let (iface, proto, name, stype, domain, host, aprot, addr, port, txt, flags) = self
```

Comment:
`let values = ...`

We only ever refer to this tuple as a whole because of the TryFrom impl below. Therefore, we only need a single variable here.